packages/opentelemetry-exporter-collector/src/types.ts

```diff
@@ -0,0 +1,482 @@
+   * If this value is 0, then no attributes were dropped.
+   */
+  droppedAttributesCount?: number;
+}
+
+/**
+ * The value of an Attribute.
+ */
+export interface AttributeValue {
+  /**
+   * A string up to 256 bytes long.
+   */
+  stringValue?: TruncatableString;
+  /**
+   * A 64-bit signed integer. May be sent to the API as either number or string
+   * type (string is needed to accurately express some 64-bit ints).
+   */
+  intValue?: string | number;
+  /**
+   * A Boolean value represented by `true` or `false`.
+   */
+  boolValue?: boolean;
+  /**
+   * A double precision floating point value.
+   */
+  doubleValue?: number;
+}
+
+/**
+ * Format for an HTTP/JSON request to a grpc-gateway for a trace span exporter.
+ */
+export interface ExportTraceServiceRequest {
+  node?: Node;
+
+  /** A list of Spans that belong to the last received Node. */
+  spans?: Span[];
+
+  /**
+   * The resource for the spans in this message that do not have an explicit
+   * resource set.
+   * If unset, the most recently set resource in the RPC stream applies. It is
+   * valid to never be set within a stream, e.g. when no resource info is known.
+   */
+  resource?: Resource;
+}
+
+/** Information on OpenTelemetry library that produced the spans/metrics. */
+export interface LibraryInfo {
+  /** Language of OpenTelemetry Library. */
+  language?: LibraryInfoLanguage;
+
+  /** Version of collector exporter of Library. */
+  exporterVersion?: string;
+
+  /** Version of OpenTelemetry Library. */
+  coreLibraryVersion?: string;
+}
+
+/**
+ * A pointer from the current span to another span in the same trace or in a
+ * different trace. For example, this can be used in batching operations, where
+ * a single batch handler processes multiple requests from different traces or
+ * when the handler receives a request from a different project.
+ */
+export interface Link {
+  /**
+   * A unique identifier for a trace. All spans from the same trace share the
+   * same `trace_id`. The ID is a 16-byte array.
+   */
+  traceId?: string;
+  /**
+   * A unique identifier for a span within a trace, assigned when the span is
+   * created. The ID is an 8-byte array.
+   */
+  spanId?: string;
+  /**
+   * The relationship of the current span relative to the linked span.
+   */
+  type?: LinkType;
+  /**
+   * A set of attributes on the link.
+   */
+  attributes?: Attributes;
+}
+
+/**
+ * A collection of links, which are references from this span to a span in the
+ * same or different trace.
+ */
+export interface Links {
+  /**
+   * A collection of links.
+   */
+  link?: Link[];
+  /**
+   * The number of dropped links after the maximum size was enforced. If this
+   * value is 0, then no links were dropped.
+   */
+  droppedLinksCount?: number;
+}
+
+/**
+ * The relationship of the two spans is unknown, or known but other than
+ * parent-child.
+ */
+export type LinkTypeUnspecified = 0;
```

Comment:
Any reason this is not an enum?

```typescript
export const enum LinkType {
  Unspecified,
  ChildLinkedSpan,
  ParentLinkedSpan,
}
```

Reply:
If we do that, could we still explicitly set the values, such as:
```ts
export const enum LinkType {
  Unspecified = 0,
  ChildLinkedSpan = 1,
  ParentLinkedSpan = 2,
}
```

(The answer may be that I didn't do that in OpenCensus code because I didn't know about const enums ....)

Reply:
@dyladan the reason is that it was ported, yes this can be an enum.

Reply:
@draffensperger 0,1,2,... is the default value for enums so you don't need `=`

Reply:
Yeah, I was thinking that the value in assigning explicit numbers was to communicate that the API expects these exact values, and so that e.g. you can't add a new value in the middle without breaking the API. But there's always a trade-off between being concise and explicit, so I'm open to either way!